src/core/default_appearance.js

```diff
@@ -0,0 +1,82 @@
+/* Copyright 2020 Mozilla Foundation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { ColorSpace } from "./colorspace.js";
+import { escapePDFName } from "./core_utils.js";
+import { EvaluatorPreprocessor } from "./evaluator.js";
+import { Name } from "./primitives.js";
+import { OPS } from "../shared/util.js";
+import { StringStream } from "./stream.js";
+
+class DefaultAppearanceEvaluator extends EvaluatorPreprocessor {
+  constructor(str) {
+    const stream = new StringStream(str);
+    super(stream, null);
+  }
+
+  parse() {
+    const operation = {
+      fn: 0,
+      args: [],
+    };
+    const result = {
+      fontSize: 0,
+      fontName: Name.get(""),
+      fontColor: new Uint8ClampedArray([0, 0, 0]) /* black */,
+    };
+
+    while (this.read(operation)) {
+      const { fn, args } = operation;
+      switch (fn | 0) {
+        case OPS.setFont:
+          result.fontSize = args[1];
+          result.fontName = args[0];
```

Comment:
Can we re-order these lines, since it feels slightly more logical accessing the array elements in order.

Reply:
Also, given that the rest of the font-handling in `src/core/evaluator.js` normally takes care of (indirectly) validating these parameters, you probably want to check here that they are of the expected types (`Name` respectively a positive number).